File: Selnium script for testing/Script with Python/test_scenario_51.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

def execute_test_51():
    """
    Dummy test run for scenario 51
    """
    driver = webdriver.Chrome()
    driver.implicitly_wait(10)
    driver.get("https://dummy.example.com/page_51")

    assert driver.find_element(By.ID, "success_msg").is_displayed()
    driver.find_element(By.XPATH, "//button[@type='submit']").click()
    driver.find_element(By.ID, "user_name").send_keys("test_data_156")
    driver.find_element(By.CLASS_NAME, "nav-item").click()
    driver.find_element(By.LINK_TEXT, "Log Out").click()
    driver.find_element(By.PARTIAL_LINK_TEXT, "Profile").click()
    driver.find_element(By.ID, "checkout_btn").click()
    driver.find_element(By.PARTIAL_LINK_TEXT, "Profile").click()
    driver.find_element(By.NAME, "password_field").send_keys("test_data_978")
    driver.find_element(By.CLASS_NAME, "nav-item").click()

    driver.quit()

if __name__ == "__main__":
    execute_test_51()

File: Selnium script for testing/Script with Java/src/test/java/testscenario/TestScenario51.java
package testscenario;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import java.time.Duration;

public class TestScenario51 {
    public static void executeTest51() {
        /*
         * Dummy test run for scenario 51
         */
        WebDriver driver = new ChromeDriver();
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        driver.get("https://dummy.example.com/page_51");

        try {
            driver.findElement(By.id("user_name")).sendKeys("test_data_416");
            driver.findElement(By.className("nav-item")).sendKeys("test_data_766");
            if (!driver.findElement(By.xpath("//button[@type='submit']")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.cssSelector(".promo-code")).click();
            if (!driver.findElement(By.className("nav-item")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.name("password_field")).sendKeys("test_data_495");
            if (!driver.findElement(By.name("password_field")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.name("password_field")).click();
            driver.findElement(By.linkText("Log Out")).sendKeys("test_data_605");
            driver.findElement(By.id("user_name")).sendKeys("test_data_386");
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            driver.quit();
        }
    }
    
    public static void main(String[] args) {
        executeTest51();
    }
}
